cmd/osctl/cmd/logs.go

```diff
@@ -47,22 +51,131 @@ var logsCmd = &cobra.Command{
 				helpers.Fatalf("error fetching logs: %s", err)
 			}
 
-			for {
-				data, err := stream.Recv()
-				if err != nil {
-					if err == io.EOF || status.Code(err) == codes.Canceled {
-						return
+			lastNode := ""
+
+			respCh, errCh := newLineSlicer(stream)
+
+			for data := range respCh {
+				if data.Metadata != nil && data.Metadata.Hostname != lastNode {
+					if lastNode != "" {
+						_, err = os.Stdout.Write([]byte("\n"))
+						helpers.Should(err)
 					}
-					helpers.Fatalf("error streaming logs: %s", err)
+
+					lastNode = data.Metadata.Hostname
+
+					_, err = os.Stdout.Write([]byte(fmt.Sprintf("==> %s <==\n", lastNode)))
```

Comment:
I'm guessing they're about equivalent, but would `fmt.Fprintf` be more appropriate here?

Reply:
I left it the way it was (direct write), but yep, I agree, given that we do tons of formatting, `fmt.Printf` might be more appropriate